### PR TITLE
Add Grunt and other updates

### LINKED ISSUE #2: Eliminate filter addition in class constructor
https://github.com/xwp/wp-foo-bar/blob/14b7abe860840a7b74da10156a2e80e24faec06e/php/class-plugin.php#L13-L31

Per https://xwp.github.io/engineering-best-practices/php/

### LINKED ISSUE #5: There is no .jshint file in dev-lib

### PR DESCRIPTION
**Pull request by valendesigns**
This PR adds the Grunt tasks that I've been using with new projects, overhauls the tests, adds `contributing.md`, closes #5, and closes #2.

### CODE REVIEW

**Comment by valendesigns** [comment]
I've tested that everything is working correctly by adding `git checkout -b feature/grunt origin/feature/grunt` to line 45 in `init-plugin.sh` and running `./init-plugin.sh "Customize Concurrency"` which I've committed and pushed up to `wp-customize-concurrency`. I had to obviously destroy the `.git` directory and re-initialize the repo and submodule due to the fact it should not create a new branch in this process or have to merge into master. I'm sure there was probably a better way, but I rather get started working on the plugin than worry about streamlining a one-off commit.

**Comment by valendesigns** [comment]
Before this is merged we should turn on Travis & Coveralls so the badges work. I've updated the tests to start off with more coverage of the core functionality. :wink:

**Comment by westonruter** [comment]
I just enabled Travis.

File: .travis.yml [hunk 1]
```diff
@@ -5,28 +5,22 @@ language:
...
-        - master
+    - WP_VERSION=latest WP_MULTISITE=1
+    - WP_VERSION=trunk WP_MULTISITE=0
+    - WP_VERSION=trunk WP_MULTISITE=1
 
 before_script:
     - export DEV_LIB_PATH=dev-lib
     - if [ ! -e "$DEV_LIB_PATH" ] && [ -L .travis.yml ]; then export DEV_LIB_PATH=$( dirname $( readlink .travis.yml ) ); fi
     - source $DEV_LIB_PATH/travis.before_script.sh
-    - export LIMIT_TRAVIS_PR_CHECK_SCOPE=patches
```
Comment (westonruter) [inline .travis.yml]: @valendesigns why remove this?

Comment (westonruter) [inline .travis.yml]: Ah, well it isn't necessary when starting a new project.

**Pull request merged**